src/components/organize/TagsManager/utils.ts

```diff
@@ -26,3 +34,45 @@ export const groupTags = (
     };
   }, {});
 };
+
+/**
+ * Returns a function for creating a tag using react-query mutations while
+ * supporting the ability to create a tag group.
+ *
+ * Provide a `createTagMutation` because this can be different for universal tags or person tags.
```

Comment:
With Gen3 we are abandoning the idea of "person tags" for universal tags. The only reason the tags are still under `/orgs/{org_id}/people/tags` is for backwards compatibility, although I realize it can be confusing.

Even the tags added to things like journeys will be created and maintained on that path (`/orgs/{org_id}/people/tags`) until we can release a v2 of the API and move them to `/orgs/{org_id}/tags`.

The tags and tag groups are still the same for every type of data, i.e. universal, which is why the new tag groups feature (where we do not have to care about backwards compatibility) is under the more universal `/orgs/{org_id}/tag_groups` path.

I don't know if all of this means that the injected `createTagMutation` could just be inlined instead?

Reply:
Oh I didn't realise we will continue to use the people tags endpoint! I will get rid of this flexible function, or maybe just use it as a hook to encapsulate the code for the purpose of handling creating a new group vs not creating a new group.

Reply:
I will leave that in a hook because dealing with conditional group creation is kinda complex and will be repeated on the journeys page.